### PR TITLE
HPCC-24536 Suppress some verbose Thor logging by default

### PR DESCRIPTION
**Pull request by jakesmith**
Allow it to be turned back on per workunit if necessary,
via #option('maxLogDetail', <value>);

Where low priority logging has been changed to logging level 200.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: thorlcr/master/thgraphmanager.cpp [hunk 1]
```diff
@@ -711,6 +711,12 @@ void CJobManager::run()
         {
             factory.setown(getWorkUnitFactory());
             workunit.setown(factory->openWorkUnit(wuid));
+
+            unsigned maxLogDetail = workunit->getDebugValueInt("maxlogdetail", DefaultDetail);
```
Comment (mckellyln) [inline thorlcr/master/thgraphmanager.cpp:715]: Does lowercase matter here ?  Typically wouldn't it be maxLogDetail ?

File: thorlcr/msort/tsortm.cpp [hunk 2]
```diff
@@ -124,7 +120,7 @@ class CSortNode: public SortSlaveMP
         try {
 
             StringBuffer epstr;
-            ActPrintLog(activity, "Connect to %s:%d",endpoint.getIpText(epstr).str(),(unsigned)mpport);
+            ActPrintLog(activity, thorDetailedLogLevel, "Connect to %s:%d",endpoint.getIpText(epstr).str(),(unsigned)mpport);
```
Comment (mckellyln) [inline thorlcr/msort/tsortm.cpp:123]: ::ActPrintLog() ?

File: thorlcr/msort/tsortm.cpp [hunk 3]
```diff
@@ -688,7 +642,6 @@ class CSortMaster : public IThorSorterMaster, public CSimpleInterface
             }
         } afor3(slaves, splitMap, numnodes, numsplits);
         afor3.For(numnodes, 20, true);
-#ifdef _TRACE
 #ifdef TRACE_PARTITION
```
Comment (mckellyln) [inline thorlcr/msort/tsortm.cpp:645]: if (!REJECTLOG(MCthorDetailedDebugInfo)) before the #ifdef here ?

File: thorlcr/msort/tsortm.cpp (hunk 2 above)
Comment (jakesmith) [reply thorlcr/msort/tsortm.cpp:123]: only needed if ambiguous with class method, there is none in this context.

File: thorlcr/master/thgraphmanager.cpp (hunk 1 above)
Comment (jakesmith) [reply thorlcr/master/thgraphmanager.cpp:715]: no, workunit debug values and the call, are case insensitive

File: thorlcr/msort/tsortm.cpp (hunk 3 above)
Comment (jakesmith) [reply thorlcr/msort/tsortm.cpp:645]: I didn't change every existing compile time directive that controlled debug tracing like this.
There is a bit of overhead even with REJECTLOG.
Could change, but don't think it's warranted for this and most other compile time section of tracnig.

**Review by mckellyln — APPROVED** [review]

**Comment by mckellyln** [comment]
@jakesmith looks good.
Many changes but almost all are mechanical.

**Comment by jakesmith** [comment]
@ghalliday - please check and merge.

File: thorlcr/thorutil/thormisc.hpp [hunk 4]
```diff
@@ -409,6 +412,14 @@ inline void ActPrintLog(const CGraphElementBase *container, const char *format,
     ActPrintLogArgs(container, thorlog_ecl, MCdebugProgress, format, args);
     va_end(args);
 }
+inline void ActPrintLog(const CGraphElementBase *container, unsigned traceLevel, const char *format, ...) __attribute__((format(printf, 3, 4)));
```
Comment (ghalliday) [inline thorlcr/thorutil/thormisc.hpp:415]: Is this function, and the GraphPrintLog function actually used?

Comment (jakesmith) [reply thorlcr/thorutil/thormisc.hpp:415]: yes, both are used.

**Pull request merged**